lib/Controller/Display.php

```diff
@@ -1444,6 +1460,101 @@ public function membershipForm(Request $request, Response $response, $id)
+     * @throws NotFoundException
+     * @throws \Xibo\Support\Exception\ControllerNotImplemented
+     */
+    public function setBandwidthLimitMultipleForm(Request $request, Response $response)
+    {
+        $sanitizedParams = $this->getSanitizer($request->getParams());
+
+        // Check if the array of ids is passed
+        if($sanitizedParams->getString('ids') == '') {
+            throw new InvalidArgumentException(__('The array of ids is empty!'));
+        }
+
+        // Get array of ids
+        $ids = $sanitizedParams->getString('ids');
+
+        $this->getState()->template = 'display-form-set-bandwidth';
+        $this->getState()->setData([
+            'ids' => $ids,
+            'help' =>  $this->getHelp()->link('Display', 'setBandwidthLimit')
+        ]);
+
+        return $this->render($request, $response);
+    }
+
+        /**
+     * Set Bandwidth to one or more displays
+     * @param Request $request
+     * @param Response $response
+     * @return \Psr\Http\Message\ResponseInterface|Response
+     * @throws AccessDeniedException
+     * @throws GeneralException
+     * @throws NotFoundException
+     * @throws \Xibo\Support\Exception\ControllerNotImplemented
+     */
+    public function setBandwidthLimitMultiple(Request $request, Response $response)
+    {
+        $sanitizedParams = $this->getSanitizer($request->getParams());
+
+        // Get array of ids
+        $ids = ($sanitizedParams->getString('ids') != '') ? explode(',', $sanitizedParams->getString('ids')) : [];
+        $bandwidthLimit = intval($sanitizedParams->getString('bandwidthLimit'));
+        $bandwidthLimitUnits = $sanitizedParams->getString('bandwidthLimitUnits');
+
+        // Check if the array of ids is passed
+        if (count($ids) == 0) {
+            throw new InvalidArgumentException(__('The array of ids is empty!'));
+        }
+
+        // Check if the bandwidth value has something
+        if ($bandwidthLimit == '') {
+            throw new InvalidArgumentException(__('The array of ids is empty!'));
+        }
+
+        foreach ($ids as $id) {
+            // get display
+            $display = $this->displayFactory->getById($id);
+
+            // check if the display is accessible by user
+            if (!$this->getUser()->checkViewable($display)) {
+                throw new AccessDeniedException();
+            }
+
+            // get respective display group
+            $displayGroup = $this->displayGroupFactory->getById($display->displayGroupId);
+
+            // convert bandwidth to kb based on form units
+            if ($bandwidthLimitUnits == 'mb') {
+                $bandwidthLimit = $bandwidthLimit * 1024;
+            } else if ($bandwidthLimitUnits == 'gb') {
+                $bandwidthLimit = $bandwidthLimit * 1024 * 1024;
+            }
+
+            // save display group
+            $displayGroup->bandwidthLimit = $bandwidthLimit;
+            $displayGroup->save(['validate' => false, 'audit' => false]);
```

Comment:
So I think in the grand scheme of things this doesn't really save us much in terms of processing, because we're still saving each one individually. It might even be slightly worse because we're adding multiple updates to one big transaction :-)

What we need is a method on the displayGroupFactory called `setBandwidth` which takes an array of displayGroupId's and then does a single update statement for all of them.

Equally I think we'd want a single audit message indicating bandwidth limits were modified on a batch of display groups at the same time.

If you'd like some direction on this, please let me know.